raiden_contracts/contracts/TokenNetwork.sol

```diff
@@ -204,6 +210,70 @@ contract TokenNetwork is Utils {
         require(participant_state.deposit >= added_deposit);
     }
 
+    /// @notice Allows `participant` to withdraw tokens from the channel that he has with
+    /// `partner`, without closing it. Can be called by anyone.
+    /// @param participant Channel participant, who will receive the withdrawn amount.
+    /// @param total_withdraw Total amount of tokens that are marked as withdrawn from the channel
+    /// during the channel lifecycle.
+    /// @param partner Channel partner address, needed to compute the channel identifier.
+    /// @param participant_signature Participant's signature on the withdraw data.
+    /// @param partner_signature Partner's signature on the withdraw data.
+    function withdraw(
+        address participant,
+        uint256 total_withdraw,
+        address partner,
+        bytes participant_signature,
+        bytes partner_signature
+    )
+        external
+    {
+        require(total_withdraw > 0);
+
+        bytes32 channel_identifier;
+        uint256 total_deposit;
+        uint256 current_withdraw;
+
+        channel_identifier = getChannelIdentifier(participant, partner);
+        Channel storage channel = channels[channel_identifier];
+
+        Participant storage participant_state = channel.participants[participant];
+        Participant storage partner_state = channel.participants[partner];
+
+        total_deposit = participant_state.deposit + partner_state.deposit;
+        current_withdraw = total_withdraw - participant_state.withdrawn_amount;
+
+        participant_state.withdrawn_amount += current_withdraw;
+
+        // Do the tokens transfer
+        require(token.transfer(participant, current_withdraw));
+
+        // Channel must be open
+        require(channel.state == 1);
+
+        verifyWithdrawSignatures(
+            channel_identifier,
+            participant,
+            partner,
+            total_withdraw,
+            participant_signature,
+            partner_signature
+        );
+
+        // Underflow check
+        require(participant_state.withdrawn_amount >= current_withdraw);
+        require(participant_state.withdrawn_amount == total_withdraw);
+
+        // Entire withdrawn amount must not be bigger than the entire channel deposit
+        require(participant_state.withdrawn_amount <= (total_deposit - partner_state.withdrawn_amount));
+
+        assert(total_deposit >= participant_state.deposit);
+        assert(total_deposit >= partner_state.deposit);
+        assert(participant_state.nonce == 0);
```

Comment:
`nonce > 0` happens only if the channel participant has a balance proof stored. This happens when closing the channel. 
I know we already have a `state` check. This is an additional check that should never fail. May be redundant, but as a general rule, we should not allow token withdrawal if we already have balance proofs stored - in case we change the protocol somehow or there's a bug.

Reply:
I see. So yes from looking at the code again this can only happen if `updateBalanceProof()` is called, which is done only during `closeChannel()`. There is a already a check for the channel state up there, requiring it to be open.

So this can never happen. So can be an assert. Funny how asserts can only be redundant checks....